tools/dbTools.py

```diff
@@ -37,13 +38,27 @@ def is_scraped(url):
     return result if result else False
 
 
-def insert_article(title, url):
+def check_file(file_hash):
+        #  Init sqlite db to filter PDFs
+        connection = sqlite3.connect('db.sqlite3')
+        cursor = connection.cursor()
+
+        cursor.execute(
+            "SELECT id FROM article WHERE file_hash = ?",
+            (file_hash,)
+        )
+        result = cursor.fetchone()
+        connection.close()
+        return result if result else False
+
+
+def insert_article(title, file_hash, url):
     #  Init sqlite db to filter PDFs
     connection = sqlite3.connect('db.sqlite3')
     cursor = connection.cursor()
     cursor.execute(
-        "INSERT INTO article (title, url) VALUES (?, ?)",
-        (title, url)
+        "INSERT INTO article (title, file_hash, url) VALUES (?, ?, ?)",
+        (title, file_hash, url)
```

Comment:
what happens if the insert fails? do you need to log errors?

Reply:
Good point. I assumed that at this point every exception should have been catch, but it's wrong. Will update 👍